contracts/strategy/LiquidationStrategy.sol

```diff
@@ -144,80 +163,140 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
+
+        _handleLogicForProfitOrLoss(fxdReceived, costOfCollateralInFXD);
+    }
+
+
+    /// @notice Allows the strategy manager to sell Collateral, held by the contract, to UniV3.
+    /// @dev Only the current strategy manager can call this function.
+    /// This function will work only when the UniversalRouter and Permit2 contracts addresses are already set in uniswapV3Info.
+    /// This allows for managing the Collateral obtained through liquidations.
+    /// @param _universalRouter The UniswapV3(or a fork)'s UniversalRouter  used for the sale.
+    /// @param _amount The amount of Collateral to sell.
+
+    function sellCollateralV3(address _collateral, address _universalRouter, uint256 _amount) external onlyStrategyManager {
+        if (_universalRouter == address(0)) revert ZeroAddress();
+        if (_amount == 0) revert ZeroAmount();
+        if (_amount > idleCollateral[_collateral].CollateralAmount) revert WrongAmount();
+        if (uniswapV3Info[_universalRouter].universalRouter == address(0) || 
+            uniswapV3Info[_universalRouter].permit2 == address(0) || 
+            uniswapV3Info[_universalRouter].poolFee == 0) revert V3InfoNotSet();
+
+        // Calculate the cost of Collateral in terms of FXD using the average price
+        uint256 costOfCollateralInFXD = _amount.mul(idleCollateral[_collateral].averagePriceOfCollateral).div(WAD); 
+        uint256 balanceOfFXDBeforeSwap = fathomStablecoin.balanceOf(address(this));
+        
+        // Existing logic for selling Collateral
+        idleCollateral[_collateral].CollateralAmount -= _amount;
+
+        _nullifyIdleCollateral(_collateral);
+
+        _adjustAmountNeededToPayDebt(_collateral, _amount.mul(idleCollateral[_collateral].averagePriceOfCollateral).div(WAD));
+
+        uint256 receivedAmount = _sellCollateralV3(_collateral, address(fathomStablecoin), _amount, uniswapV3Info[_universalRouter].poolFee, _universalRouter);
+        emit LogSellCollateralV3(_universalRouter, _amount, receivedAmount); // This line is crucial for logging the sale details
+
+        uint256 balanceOfFXDAfterSwap = fathomStablecoin.balanceOf(address(this));
+        uint256 fxdReceived = balanceOfFXDAfterSwap.sub(balanceOfFXDBeforeSwap); // FXD gained from the swap
+        
+        _handleLogicForProfitOrLoss(fxdReceived, costOfCollateralInFXD);
     }
 
-    /// @notice Handles the liquidation process, swapping WXDC for FXD, and repaying debt.
+
+    /// @notice Withdraws a specified amount of Collateral from the contract.
+    /// @dev Only the current strategy manager can call this function.
+    /// Useful for managing the reserves after liquidation events.
+    /// @param _amount The amount of Collateral to withdraw.
+    function shutdownWithdrawCollateral(address _collateral, uint256 _amount) external onlyStrategyManager {
+        if (_amount == 0) revert ZeroAmount();
+        if (_amount > idleCollateral[_collateral].CollateralAmount) revert WrongAmount();
```

Comment:
we should store idleCollateral[_collateral].CollateralAmount in local variable here